OrbitCore/EventBuffer.h

```diff
@@ -15,43 +15,66 @@
 
 class EventBuffer {
  public:
-  EventBuffer() : m_MaxTime(0), m_MinTime(LLONG_MAX) {}
+  EventBuffer() : max_time_(0), min_time_(std::numeric_limits<uint64_t>::max()) {}
 
   void Reset() {
     callstack_events_.clear();
-    m_MinTime = LLONG_MAX;
-    m_MaxTime = 0;
+    min_time_ = std::numeric_limits<uint64_t>::max();
+    max_time_ = 0;
   }
-  std::map<ThreadID, std::map<uint64_t, orbit_client_protos::CallstackEvent> >& GetCallstacks() {
+
+  [[nodiscard]] const std::map<int32_t, std::map<uint64_t, orbit_client_protos::CallstackEvent>>&
+  GetCallstacks() const {
     return callstack_events_;
   }
-  Mutex& GetMutex() { return m_Mutex; }
-  std::vector<orbit_client_protos::CallstackEvent> GetCallstackEvents(
+
+  [[nodiscard]] const std::map<uint64_t, orbit_client_protos::CallstackEvent>&
+  GetCallstacksOfThread(int32_t thread_id) const {
+    static std::map<uint64_t, orbit_client_protos::CallstackEvent> empty;
+    const auto& it = callstack_events_.find(thread_id);
+    if (it == callstack_events_.end()) {
+      return empty;
+    }
+    return it->second;
+  }
+
+  Mutex& GetMutex() { return mutex_; }
+
+  [[nodiscard]] std::vector<orbit_client_protos::CallstackEvent> GetCallstackEvents(
       uint64_t time_begin, uint64_t time_end,
-      ThreadID thread_id = SamplingProfiler::kAllThreadsFakeTid);
-  uint64_t GetMaxTime() const { return m_MaxTime; }
-  uint64_t GetMinTime() const { return m_MinTime; }
-  bool HasEvent() {
-    ScopeLock lock(m_Mutex);
-    return !callstack_events_.empty();
+      int32_t thread_id = SamplingProfiler::kAllThreadsFakeTid) const;
+
+  [[nodiscard]] uint64_t GetMaxTime() const { return max_time_; }
+
+  [[nodiscard]] uint64_t GetMinTime() const { return min_time_; }
+
+  [[nodiscard]] bool HasEvent() {
+    ScopeLock lock(mutex_);
+    if (callstack_events_.empty()) {
+      return false;
+    }
+    for (const auto& pair : callstack_events_) {
+      if (!pair.second.empty()) {
+        return true;
+      }
+    }
+    return true;
```

Comment:
Shouldn't this be `return false` as all the maps were found empty?

Reply:
And that's why code reviews are so useful (and test would be even better.. :)) Thanks for spotting this!

Reply:
just updated the branch again.